packages/aquedux-client/src/createClient.js

```diff
@@ -0,0 +1,122 @@
+// @flow
+
+import SockJS from 'sockjs-client'
+
+import actionTypes from './constants/actionTypes'
+import configManager, { type AqueduxConfig } from './managers/configManager'
+import { raise, register, unregister, events } from './utils/eventHub'
+import localStorage from './utils/localStorage'
+
+export default function createClient(options: AqueduxConfig) {
+  const { endpoint, hydratedActionTypes } = configManager.setConfig(options)
+
+  let socket: ?SockJS = null
+  let actionStack: Array<Object> = []
+
+  // Ideally, these actions should not be sent back to client. Duplicates definition in middleware
+  function isWiredActionType(actionType: string): boolean {
+    const internalActionTypes = [
+      actionTypes.AQUEDUX_CLIENT_CHANNEL_JOIN,
+      actionTypes.AQUEDUX_CLIENT_CHANNEL_LEAVE,
+      actionTypes.AQUEDUX_CHANNEL_SNAPSHOT
+    ]
+
+    return internalActionTypes.indexOf(actionType) !== -1 || hydratedActionTypes.indexOf(actionType) !== -1
+  }
+
+  function onSend(action) {
+    // If the socket is not open, we stack the actions.
+    if (!socket || socket.readyState !== 1) {
+      actionStack.push(action)
+      return
+    }
+
+    // Append our jwt from the local storage to the action before sending it.
+    const token = localStorage.getItem('aquedux-jwt')
+    if (token !== null) {
+      action = {
+        ...action,
+        token
```

Comment:
Maybe better to keep `aquedux-jwt` in action to prevent a name collision.

Reply:
Do you mean ?
```js
const token = localStorage.getItem('aquedux-jwt')
    if (token !== null) {
      action = {
        ...action,
        'aquedux-jwt': token
      }
    }
```
If so I'd have to change some code in aquedux-server too in this PR. This may be resolved in another issue? Also a generic key for any aquedux-related informations (roundtrip timestamps in the future) could be a nice approach:
```js
{
   aquedux: {
      token,
      // receivedTimestamp,
      // sentTimestamp
   }
}
```
PS: Ignore the commented keys here. Those are just here for a mind representation of the future key shape and as an excuse to turn the token into an object ;)

Reply:
Let's do this

Reply:
Issue #37 created ;)